source/use-cases/sensitive-data-encryption.txt

```diff
@@ -57,12 +57,119 @@ cubilia Curae; Nunc non interdum purus, ultricies laoreet tortor.
-porta.
+While designing your database, you may want to enforce particular datatypes in specific fields. MongoDB uses JSON Schema to reduce the complexity of data validation during insert and update operations. Field Level Encryption with MongoDB requires the use of JSON Schema to specify several configuration options, allowing the drivers to encrypt and decrypt fields automatically.
+
+* With CSFLE enabled, the driver automatically encrypts and decrypted fields that have the following configuration options specified:
+    - The encryption algorithm (deterministic or random)
+    - The data key
+    - The BSON Type
+
+.. note::
+
+    A single data key can be used for all encrypted fields, or you can use individual data keys that are specific to each field.
+
+
+The MedcoMD engineers receive specific requirements for the fields of data. `Deterministic Encryption <https://docs.mongodb.com/manual/core/security-client-side-encryption/#deterministic-encryption>`_ and `Random encryption <https://docs.mongodb.com/manual/core/security-client-side-encryption/#random-encryption>`_ are the two different methods of encrypting sensitive fields. The following table illustrates the data model of the Medco Management System.
+
+.. table:: 
+    :widths: 50 50
+
+    +-----------------------------+---------------------+----------------------------------------------+
+    |Field type                   |Encryption Algorithm | BSON Type                                    |
+    +=============================+=====================+==============================================+
+    | Name                        | Non-Encrypted       | String                                       |
+    +-----------------------------+---------------------+----------------------------------------------+
+    | SSN                         | Deterministic       |  Int                                         |
+    +-----------------------------+---------------------+----------------------------------------------+
+    | Blood Type                  | Random              |  String                                      |
+    +-----------------------------+---------------------+----------------------------------------------+
+    | Medical Records             | Random              |  Array                                       |
+    +-----------------------------+---------------------+----------------------------------------------+
+    | Insurance: Policy Number    | Deterministic       |  Int (embedded inside `insurance` object)    |
+    +-----------------------------+---------------------+----------------------------------------------+
+    | Insurance: Provider         | Non-Encrypted       |  String (embedded inside `insurance` object) |
+    +-----------------------------+---------------------+----------------------------------------------+
+
+Pass the schema as a JSON object to the secure client to specify the encrypted field properties.
```

Comment:
This transition is abrupt -- sorry if I missed that while doing a spot check in another review. In the prior sentence, you describe the Medco Management System data model, and then here, ask the reader to do something with JSON schema without connecting the two. You could try rearranging some of the content you have, e.g.:
```
line 93:
In order to ensure fields are encrypted with the appropriate encryption algorithm, data key, and BSON type, create a JSON Schema using the information in the data model and pass it to the secure client as shown below:


.. note::
   The encryptMetadata is used in this example to allow the encryption-specific properties to be inherited by fields defined in the properties section.
```

Reply:
Some things I agree with in your comment:
* The transition is abrupt
* I haven't connected the previous bit (data model) with the following bit (json schema config)

Some things I don't agree with and why:
* Specifying that we're using an encryption algorithm, data key, and BSON Type. I don't agree with this because we've already mentioned those 3 things that we need and it's demonstrated right below, at that point mentioning it here would be repetitive.
* Putting a note right after we say 'pass it to the secure client as shown below:' because it's a continuity issue where we say we're doing something below and then show a note instead of the thing we said we're going to show.

Here's what I chose to replace it with to address the transition and connection issue:
"
To automatically encrypt and decrypt create a JSON Schema using the information in the data model and pass the schema as a JSON object to the secure client to specify the encrypted field properties.
In the JSON, you can specify `encryptMetadata <https://docs.mongodb.com/manual/reference/security-client-side-automatic-json-schema/#encryptmetadata-schema-keyword>`_ that is passed down to the fields under the ``properties`` nested object. 
"

Reply:
Are you okay w/ resolving this discussion if you feel your concerns are addressed? Let me know if there's something left to fix w/ this comment thread

@ccho-mongodb